src/main/java/Duke.java

```diff
@@ -1,10 +1,174 @@
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        initTaskList();
+        showWelcomeMessage();
+        while (true) {
+            String command = readUserInput();
+            executeCommand(command);
+        }
+    }
+
+    private static void initTaskList() {
+        allTasks = new Task[TASK_CAPACITY];
+        count = 0;
+    }
+
+    private static void showWelcomeMessage() {
+        System.out.println(MESSAGE_WELCOME);
+    }
+
+    private static String readUserInput() {
+        return SCANNER.nextLine();
+    }
+
+    private static void executeCommand(String input) {
+        final String[] commandWordAndArgs = splitCommandWordAndArgs(input);
+        final String commandWord = commandWordAndArgs[0];
+        final String commandArgs = commandWordAndArgs[1];
+
+        switch (commandWord) {
+            case COMMAND_TODO:
+                executeAddTodo(commandArgs);
+                break;
+            case COMMAND_DEADLINE:
+                executeAddDeadline(commandArgs);
+                break;
+            case COMMAND_EVENT:
+                executeAddEvent(commandArgs);
+                break;
+            case COMMAND_LIST:
+                executeList();
+                break;
+            case COMMAND_MARK:
+                executeMark(commandArgs);
+                break;
+            case COMMAND_UNMARK:
+                executeUnmark(commandArgs);
+                break;
+            case COMMAND_EXIT:
+                executeExit();
+                break;
+            default:
+                executeError();
+                break;
+        }
+    }
+
+    private static String[] splitCommandWordAndArgs(String input) {
+        final String[] split = input.split(" ", 2);
+        return split.length == 2 ? split : new String[]{split[0], ""};
+    }
+
+    private static void executeAddTodo(String input) {
+        allTasks[count] = new Todo(input);
+        echoTask();
+    }
+
+    private static void executeAddDeadline(String input) {
+        String[] split = splitDeadlineAndDate(input);
+        allTasks[count] = new Deadline(split[0], split[1]);
+        echoTask();
+    }
+
+    private static String[] splitDeadlineAndDate(String input) {
+        final int indexOfByPrefix = input.indexOf(PREFIX_BY);
+        String[] deadlineAndDate = new String[2];
+
+        deadlineAndDate[0] = input.substring(0, indexOfByPrefix).trim();
+        deadlineAndDate[1] = removePrefix(input.substring(indexOfByPrefix), PREFIX_BY).trim();
+        return deadlineAndDate;
+    }
+
+    private static void executeAddEvent(String input) {
+        String[] split = splitEventAndDates(input);
```

Comment:
I would suggest a clearer variable name than split.
```suggestion
        String[] eventDetails= splitEventAndDates(input);
        String eventTitle = eventDetails[0];
        String eventStart = eventDetails[1];
        String eventEnd = eventDetails[2];
```